packages/snaps-sdk/src/internals/svg.ts

```diff
@@ -1,48 +1,20 @@
-import { assert, hasProperty, isObject } from '@metamask/utils';
-import { XMLParser } from 'fast-xml-parser';
+import { refine, string } from 'superstruct';
 
 /**
- * Parse and validate a string as an SVG.
+ * Get a Struct that validates a string as a valid SVG.
  *
- * @param svg - An SVG string.
- * @returns The SVG, its attributes and children in an object format.
+ * @returns A Struct that validates a string as a valid SVG.
+ * @internal
  */
-export function parseSvg(svg: string) {
-  try {
-    const trimmed = svg.trim();
-
-    assert(trimmed.length > 0);
-
-    const parser = new XMLParser({
-      ignoreAttributes: false,
-      parseAttributeValue: true,
-    });
-    const parsed = parser.parse(trimmed, true);
-
-    assert(hasProperty(parsed, 'svg'));
-
-    // Empty SVGs are not returned as objects
-    if (!isObject(parsed.svg)) {
-      return {};
+export function svg() {
+  return refine(string(), 'SVG', (value) => {
+    // This validation is intentionally very basic, we don't need to be that strict
+    // and merely have this extra validation as a helpful error if devs aren't
+    // passing in SVGs.
+    if (!value.startsWith('<svg')) {
```

Comment:
SVGs may start with an XML declaration. Maybe just check if the string contains `<svg`.

Reply:
Good call, done!